### PR TITLE
[JIT] Fix potential division by zero in fgopt.cpp

### PR DESCRIPTION
**Pull request by grospelliergilles**
Division by zero in C++ is undefined behavior on some architecture and should be avoided.

Related to #76257

### CODE REVIEW

File: src/coreclr/jit/fgopt.cpp [hunk 1]
```diff
@@ -4988,7 +4988,7 @@ bool Compiler::fgReorderBlocks(bool useProfile)
                         double notTakenCount =
                             ((double)edgeToBlock->edgeWeightMin() + (double)edgeToBlock->edgeWeightMax()) / 2.0;
                         double totalCount = takenCount + notTakenCount;
-                        double takenRatio = takenCount / totalCount;
+                        double takenRatio = (totalCount!=0.0) ? (takenCount / totalCount) : 0.0;
```
Comment (AndyAyersMS) [inline src/coreclr/jit/fgopt.cpp:4991]: How about getting rid of the divide entirely?

```C++
                        // If the takenRatio is greater or equal to 51% then we will reverse the branch
                        //
                        if (takenCount <= (0.51) * totalCount)
                        {
```

Comment (grospelliergilles) [reply src/coreclr/jit/fgopt.cpp:4991]: Yes it is a very good idea. I have updated my PR with your suggestion.

**Comment by AndyAyersMS** [comment]
@grospelliergilles the jit is picky about formatting. I pushed a fix to your fork.

The SPMI replay failure is a timeout. @BruceForstall I've seen this now a few times lately.
```
['win-x86-2' END OF WORK ITEM LOG: Command timed out, and was killed]
```

**Review by AndyAyersMS — APPROVED** [review]
Thanks for noticing and fixing this!

**Pull request merged**